### PR TITLE
📖 Blog post on JupyterLab MyST

### PR DESCRIPTION
**Pull request by rowanc1**
# Read the [blog post](https://executablebookproject--1043.org.readthedocs.build/en/1043/blog/2023-06-27-jupyterlab-myst/)

I have put together a blog post specifically on some of the features for JupyterLab MyST. There is a ton in here to highlight, and as we have just released support for JupyterLab v4, it is a good time to socialize it! 🚀

@stevejpurves and I are planning a webinar on the 5th, and we will also have a poster at SciPy the following week.

@agoose77 could you take a look at the blog and see if there are any changes? One of the main changes that I would like to make is adding citation support in the next few weeks as well as improve and merge some of the user-facing docs.

@stevejpurves could you add the future link to the YouTube video as well as take a pass on any edits?!

@choldgraf and @mmcky tagging you both if you want to take a read/edit as well!

### CODE REVIEW

**Comment by rowanc1** [comment]
I pinned the video links to githubs CDN, maybe some one else knows how to ensure that the static folder is copied properly in sphinx? This did work locally, but not on RTD.

**Comment by rowanc1** [comment]
Also thanks @roaldarbol for the tips on JupyterLab Desktop -- excited about the next steps on the LaPreprint template. 🚀

File: docs/blog/2023-06-27-jupyterlab-myst.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+MyST extension in JupyterLab with a rich authoring experience, frontmatter and inline expressions.
+```
+
+To follow along with the tutorial you can install the plugin using pip and see the full [getting started guide](https://mystmd.org/guide/quickstart-jupyter-lab-myst):
+
+```shell
+pip install jupyterlab_myst
+```
+
+The plugin works best in the latest version of JupyterLab (version 4.0 or greater). We released v2 of our plugin on [June 23rd, 2023](https://github.com/executablebooks/jupyterlab-myst/releases/tag/v2.0.0), so if you have already installed the plugin, you can update to v2!
```
Comment (stevejpurves) [inline docs/blog/2023-06-27-jupyterlab-myst.md:36]: is this saying that v2 is compatible with jlab 4.0 and 3.x?

File: docs/blog/2023-06-27-jupyterlab-myst.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+- Expand a sympy equation `polynomial`: `` {eval}`expand(polynomial)` ``
+- Add inline matplotlib “sparklines”
+- Add ipywidgets directly inline for sliders and other dropdowns
+
+<figure>
+<video src="https://github.com/executablebooks/meta/raw/d30cae55226c762a956c5a33cf9fd6e93557f3f8/docs/_static/videos/jupyterlab-myst-inline.mp4" autoplay muted webkit-playsinline="true" playsinline loop style="max-width:100%"></video>
+<figcaption>Inline execution in markdown cells using the <code>{eval}</code> role.</figcaption>
+</figure>
+
+Inline expressions open up a whole new world for communicating with interactive computation, and we are excited to explore this and other integrations with [thebe](https://github.com/executablebooks/thebe) and [JupyterLite](https://jupyterlite.readthedocs.io/en/latest/) in the coming months.
```
Comment (stevejpurves) [inline docs/blog/2023-06-27-jupyterlab-myst.md:80]: ...open up a whole new world [of narrative flow] for communicating with interactive computation...

**Review by choldgraf — DISMISSED** [review]
This generally looks great to me - once @stevejpurves is happy w/ wording changes etc I'm +1 on merge. I don't think it's worth the time it'd take to figure out how to get static assets copied properly with Sphinx.

File: docs/blog/2023-06-27-jupyterlab-myst.md (hunk 1 above)
Comment (agoose77) [reply docs/blog/2023-06-27-jupyterlab-myst.md:36]: We will make v2 compatible with JupyterLab-3. We need to update our dependencies in `package.json` to reflect the compatibility, and I need to run through and check what's broken on 3.0 :P

If we can hold off on 3.0 advertisement, I woudl

File: docs/blog/2023-06-27-jupyterlab-myst.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+Join the upcoming webinar (5th July 2023 @ 3pm UTC) and get started using MyST in JupyterLab and how it connects to other scientific writing workflows. The webinar is hosted by Rowan & Steve from [Curvenote](https://curvenote.com), and will be an hour walk-through of using [MyST in JupyterLab](https://github.com/executablebooks/jupyterlab-myst).
+
+[Sign up for the webinar](https://www.eventbrite.ca/e/scientific-writing-in-jupyterlab-with-myst-markdown-tickets-666670598707) or [watch the recording on YouTube](https://youtube.com/live/12-Go9E2huQ) after.
+:::
+
+If you are using JupyterBook or MyST Markdown, you can now see your content directly in JupyterLab with the our new extension [`jupyterlab-myst`](https://github.com/executablebooks/jupyterlab-myst). In this post we will showcase `jupyterlab-myst` to enable rich parsing and rendering of MyST natively in JupyterLab, including:
+
+1. rich authoring components like figures, hover cross-references, tabbed content, and admonitions;
+2. including variables, interactive widgets, and graphs directly inside of Markdown cells in the notebook; and
+3. capturing and displaying rich frontmatter.
```
Comment (agoose77) [inline docs/blog/2023-06-27-jupyterlab-myst.md:21]:
```suggestion
1. Rich authoring components like figures, hover cross-references, tabbed content, and admonitions;
2. Including variables, interactive widgets, and graphs directly inside of Markdown cells in the notebook; and
3. Capturing and displaying rich frontmatter.
```

**Comment by agoose77** [comment]
@rowanc1 

> Rabbit-hole links for cross-referenced content

Does this refer to cross-document referencing? I didn't track any JupyterLab work to make this work (we're single document for now).

File: docs/blog/2023-06-27-jupyterlab-myst.md (hunk 1 above)
Comment (rowanc1) [reply docs/blog/2023-06-27-jupyterlab-myst.md:36]: The plugin does work in 3x series, with the exception of inline ipywidgets. Simple inline execution does work, all of the parsing does work as well!

My plan today is to do a pass on docs, and I think over there we can be a bit clearer about the exact requirements and leave this side open? (e.g. it works "best" with v4..!).

**Comment by rowanc1** [comment]
For the "Rabbit hole links" - maybe we just take that out for now. It can work for figure in in a footnote, with an equation referenced etc. But it is all in a single document!

File: docs/blog/2023-06-27-jupyterlab-myst.md (hunk 2 above)
Comment (rowanc1) [reply docs/blog/2023-06-27-jupyterlab-myst.md:80]: Based on your comment @stevejpurves:

```suggestion
Inline expressions open up a whole new world for communicating with interactive computation directly in your markdown narrative, and we are excited to explore this and other integrations with [thebe](https://github.com/executablebooks/thebe) and [JupyterLite](https://jupyterlite.readthedocs.io/en/latest/) in the coming months.
```

**Comment by rowanc1** [comment]
Thanks all for the help! Merging this in and will start promoting this in various circles! 🚀

**Review by agoose77 — APPROVED** [review]

**Comment by mmcky** [comment]
Thanks @rowanc1 this is exciting. 

I will send this around to the QuantEcon team to watch the video of it afterwards (as its 1am our time).

**Comment by rowanc1** [comment]
Will record! Making sure that Steve is awake for tag teaming the webinar! :)

**Pull request merged**